.mergify.yml

```diff
@@ -0,0 +1,37 @@
+pull_request_rules:
+  - name: automatic merge
```

Comment:
As observed in https://github.com/os-autoinst/os-autoinst/pull/1842#issuecomment-976578261, one major issue with mergify is that pushes to the pull request do not invalidate existing approvals. Thus it's possible to push broken (or even malicious) changes between review and merge.

Reply:
That's correct but IMHO has nothing to do with mergify. I think github can be configured to invalidate approvals in case any new commits change but that would defeat the purpose of allowing the author to do trivial, pre-agreed changes, like applying perltidy or fixing typos in comments or something. It's still in the responsibility of reviewers to only approve if the PR is *actually* acceptable

Reply:
> I think github can be configured to invalidate approvals in case any new commits change but that would defeat the purpose of allowing the author to do trivial, pre-agreed changes, like applying perltidy or fixing typos in comments or something.

IMO that option makes sense and should be enabled. Approvals on earlier revisions are still visible, so nothing is lost by doing that, but there is a distinction between "this state was reviewed" and "someone reviewed some older, possibly totally different code in the past", which is necessary for a proper decision by humans as well as bots.

> It's still in the responsibility of reviewers to only approve if the PR is actually acceptable

If the PR is "actually acceptable", reviewers can also hit the "Merge after pipeline succeeds" button, which is pretty much equivalent to adding an approval and waiting for mergify, just that there is no race condition which allows to subvert the review.

Reply:
> That's correct but IMHO has nothing to do with mergify. I think github can be configured to invalidate approvals in case any new commits change but that would defeat the purpose of allowing the author to do trivial, pre-agreed changes, like applying perltidy or fixing typos in comments or something. It's still in the responsibility of reviewers to only approve if the PR is _actually_ acceptable

Then you'd also have to forbid approving PRs which are marked as WIP. Without invalidating reviews on every push, the easiest way to sneak malicious code into the repo is:
- mark the PR as WIP
- wait for 2 reviews
- push malicious code
- remove WIP

You can have either automerge, or approvals that persist after code push. But not both.

Reply:
ok. With https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455/commits/e9f9a6f18852d892735a55b935617c8ac671a37c now approvals will be removed if changes have been pushed. Also now there won't be an automatic merge just based on successful checks, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455/commits/2a7b923a361fed6fe6e6e81de28d46e728c5ddc2